internals/plan/plan.go

```diff
@@ -42,11 +67,60 @@ const (
 	defaultCheckThreshold = 3
 )
 
+// layerExtensions keeps a map of registered extensions.
+var layerExtensions = map[string]LayerSectionExtension{}
+
+// RegisterExtension adds a plan schema extension. All registrations must be
+// done before the plan library is used.
+func RegisterExtension(field string, ext LayerSectionExtension) {
+	if _, ok := layerExtensions[field]; ok {
+		panic(fmt.Sprintf("internal error: extension %q already registered", field))
+	}
+	layerExtensions[field] = ext
```

Comment:
We need to panic if the name clashes with a pebble field name.

Reply:
Yes! Fixed, thank you! Added a unit test as well. Resolving.